### DISCUSSION TITLE
Possible to send multiple queries at once?

### DISCUSSION THREAD

**Discussion by ozziestraw** [pocketbase/pocketbase#1642] [Q&A]
To prevent having to send multiple HTTP requests (i.e. multiple SELECT's for different unrelated tables) - would it be possible for the API to receive multiple queries, process them independently and return everything in one JSON response?

**Comment by ganigeorgiev** [comment]
No, this is not supported at the moment. Bulk selects are not planned, but transactional create/update/delete eventually will be handled as part of #48.

**Reply by ozziestraw** [reply]
Thanks for the reply -- Bulk selects can significantly boost performance as it limits the number of requests needed -- could it not be included in the update alongside bulk create/update/delete?

**Reply by ganigeorgiev** [reply]
> could it not be included in the update alongside bulk create/update/delete?

No, sorry for now this is not planned since it is not clear how the response will be returned and evaluated when there is pagination or expand for example. Also keep in mind that sending and receiving one large request/response may not always be better and in some situations can even hurt the performance due to allocations and GC pressure since we'll have to hold all fetched data in the memory until the request is processed.
